pkg/apis/core/types_cloudprofile.go

```diff
@@ -252,6 +264,29 @@ type BastionMachineType struct {
 	Name string
 }
 
+// CloudProfileStatus contains the status of the cloud profile.
+type CloudProfileStatus struct {
+	// KubernetesVersions contains the statuses of the kubernetes versions.
+	KubernetesVersions []ExpirableVersionStatus
+	// MachineImageVersions contains the statuses of the machine image versions.
+	MachineImageVersions []MachineImageVersionStatus
+}
+
+type MachineImageVersionStatus struct {
```

Comment:
Requires docstring. See `v1beta1/types_cloudprofile.go`:
```suggestion
// MachineImageVersionStatus contains the status of a machine image and its versions.
type MachineImageVersionStatus struct {
```